mitmproxy/http.py

```diff
@@ -944,7 +944,9 @@ def _get_multipart_form(self):
 
     def _set_multipart_form(self, value):
         self.content = multipart.encode(self.headers, value)
-        self.headers["content-type"] = "multipart/form-data"
+        if "content-type" not in self.headers:
+            # Don't overwrite header if it already exists or it will destroy the boundary value
+            self.headers["content-type"] = "multipart/form-data"
```

Comment:
This... looks like it plainly didn't work before? Good catch.

I would suggest we change the logic here to 

1. Check if `self.headers["content_type"].startswith("multipart/form-data")`, and if that's not the case, add a content-type header with a random (?) boundary. The point here is that if someone assigns to `.multipart_form = ` in their addon, the header should be updated accordingly if the request was not multipart before.
2. *After* that, call `self.content = multipart.encode(self.headers, value)` (as we do right now)